src/Contracts/Str.php

```diff
@@ -0,0 +1,46 @@
+<?php
```

Comment:
What is the goal of having immutable strings? What do we expect to accomplish with it?

I understand keeping track of object state is important, and immutability helps in that aspect. But if an object is a primitive type wrapper, is its state important? 

As a side note, primitive types are always passed by copy in PHP, where objects are always passed as references. In this aspect, immutable objects replicate best a primitive type, as any modification will not propagate everywhere. The performance loss greatly concerns me, though. Benchmarks would be greatly appreciated here.